### PR TITLE
Make compatible with python3

### LINKED ISSUE #34: Make it Python 3.x compatible
We'll need it for our new project (which will be run under Python 3.x)

### PR DESCRIPTION
**Pull request by jamesodo**
Fixes #34 

Best viewed one commit at a time.

Should work with 2.7, 3.2, 3.3, 3.4, 3.5
Tested with the latest patch version of each minor version (except 2.7.3 because that is what is currently on our servers):
2.7.3
3.2.6
3.3.6
3.4.4
3.5.1

Skipped 3.0 and 3.1 because they both reached end of life at least 4 years ago.
3.2 reached end of life in Feb 2016 but is included because it is the Python 3 that is ships with Ubuntu Precise.

pip version needs to be 7.1.2 because 8.0 dropped support for Python 3.2. Wasn't sure if that should be in requirements.txt or not?

### CODE REVIEW

File: yoconfigurator/dicts.py [hunk 1]
```diff
@@ -15,7 +26,7 @@ class DotDict(dict):
 
     def __init__(self, *args, **kwargs):
         super(DotDict, self).__init__(*args, **kwargs)
-        for key, value in self.iteritems():
+        for key, value in iteritems(self):
```
Comment (thomasw) [inline yoconfigurator/dicts.py]: I would just use `.items()` here and below. IIRC, .items() does the "right thing" now in Python 3, so using items will be less performant on 2.7 but that doesn't really seem like the end of the world in this case.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1 @@
-# Only for tests on Python < 2.7:
-argparse
-unittest2
+virtualenv==13.1.2
```
Comment (stefanor) [inline requirements.txt]: I don't think that belongs here. If our tools are using too old a virtualenv, we must fix them.

File: setup.py [hunk 3]
```diff
@@ -1,12 +1,7 @@
 #!/usr/bin/env python
 
-import sys
-
 from setuptools import setup, find_packages
 
-install_requires = []
-if sys.version_info < (2, 7):
-    install_requires.append('argparse')
```
Comment (stefanor) [inline setup.py]: Now that we've got rid of this, we can add a `setup.cfg` that specifies universal wheels.

File: setup.py [hunk 4]
```diff
@@ -18,6 +13,5 @@
     version='0.4.6',
     packages=find_packages(),
     scripts=['bin/configurator.py'],
-    install_requires=install_requires,
     test_suite='yoconfigurator.tests',
```
Comment (stefanor) [inline setup.py]: Add trove classifiers for Python 3.

File: .travis.yml [hunk 5]
```diff
@@ -1,7 +1,10 @@
...
 python:
-    - "2.6"
     - "2.7"
+    - "3.2"
+    - "3.3"
+    - "3.4"
+    - "3.5"
 install:
-    - pip install -r requirements.txt --use-mirrors
+    - pip install virtualenv==13.1.2
```
Comment (thomasw) [inline .travis.yml]: Hm. I suspect that something isn't right if specifying this is necessary. I have projects with the same python dependencies where I don't need to do this. One difference though is that I'm explicitly using their contained based infrastructure. Can you add `sudo: false` to the top of this file and remove the install section just to see if there is a difference?

Comment (jamesodo) [inline .travis.yml]: Yeah, it wasn't needed. Looks like travis is configured with virtualenv and pip version old enough to work find with 3.2.

File: CHANGELOG.rst [hunk 6]
```diff
@@ -1,6 +1,11 @@
 Change Log
 ==========
 
+Dev
+-----
+
+* Update to work with Python 2.7, 3.2, 3.3, 3.4, 3.5
```
Comment (zoidbergwill) [inline CHANGELOG.rst]: And drop support for 2.6?

**Comment by thomasw** [comment]
This looks good to me, but I'll let Stefano give the final thumb.

Do you know why Jenkins isn't happy?

**Comment by stefanor** [comment]
:+1:

> Do you know why Jenkins isn't happy?

Because it explicitly installs `yconfigurator`. I don't think Jenkins can usefully test yoconfigurator, with this its usual setup.

**Pull request merged**